codebundles/k8s-kubectl-namespace-healthcheck/sli.robot

```diff
@@ -0,0 +1,114 @@
+*** Settings ***
+Metadata          Author    Shea Stewart
+Documentation     This SLI uses kubectl to score namespace health. Produces a value between 0 (completely failing thet test) and 1 (fully passying the test). Looks for container restarts, events, and pods not ready.
+Force Tags        K8s    Kubernetes    Kube    K8    Namespace    Health
+Suite Setup       Suite Initialization
+Library           BuiltIn
+Library           RW.Core
+Library           RW.Utils
+Library           RW.K8s
+Library           RW.platform
+Library           OperatingSystem
+
+*** Keywords ***
+Suite Initialization
+    ${kubeconfig}=    RW.Core.Import Secret    kubeconfig
+    ...    type=string
+    ...    description=The kubernetes kubeconfig yaml containing connection configuration used to connect to cluster(s).
+    ...    pattern=\w*
+    ...    example=For examples, start here https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/
+    ${kubectl}=    RW.Core.Import Service    kubectl
+    ...    description=The location service used to interpret shell commands.
+    ...    default=kubectl-service.shared
+    ...    example=kubectl-service.shared
+    ${NAMESPACE}=    RW.Core.Import User Variable    NAMESPACE
+    ...    type=string
+    ...    description=The name of the Kubernetes namespace to scope actions and searching to. Supports csv list of namespaces, or ALL. 
+    ...    pattern=\w*
+    ...    example=ALL
+    ${CONTEXT}=    RW.Core.Import User Variable    CONTEXT
+    ...    type=string
+    ...    description=Which Kubernetes context to operate within.
+    ...    pattern=\w*
+    ...    example=my-main-cluster
+    ${DISTRIBUTION}=    RW.Core.Import User Variable    DISTRIBUTION
+    ...    type=string
+    ...    description=Which distribution of Kubernetes to use for operations, such as: Kubernetes, OpenShift, etc.
+    ...    pattern=\w*
+    ...    enum=[Kubernetes,GKE,OpenShift]
+    ...    example=Kubernetes
+    ${EVENT_TYPE}=    RW.Core.Import User Variable    EVENT_TYPE
+    ...    type=string
+    ...    description=The error pattern to use when grep-ing logs or searching events.
+    ...    pattern=\w*
+    ...    example=Normal|Warning
+    ...    default=Warning
+    ${EVENT_AGE}=    RW.Core.Import User Variable    EVENT_AGE
+    ...    type=string
+    ...    description=The time window in minutes as to when the event was last seen.
+    ...    pattern=\w*
+    ...    example=5m
+    ...    default=5m
+    ${EVENT_THRESHOLD}=    RW.Core.Import User Variable    EVENT_THRESHOLD
+    ...    type=string
+    ...    description=The maximum total events to be still considered healthy. 
+    ...    pattern=\w*
+    ...    example=2
+    ...    default=0
+    ${CONTAINER_RESTART_AGE}=    RW.Core.Import User Variable    CONTAINER_RESTART_AGE
+    ...    type=string
+    ...    description=The time window in minutes as search for container restarts.
+    ...    pattern=\w*
```

Comment:
Depending on time units you intended to accept, something like `    ...    pattern=((\d+?)d)?((\d+?)h)?((\d+?)m)?((\d+?)s)?`

Reply:
I was really only planning on accepting minutes here. 
a) the time calculation function currently only supports minutes, 
b) the time calculation function was originally built for searching events, which are only ever stored by the kubeapi for 1 hour (so minutes seemed like a fine value to accept)
c) I don't see value in counting container restarts over any length of time more than an hour (and really for me, only 5 minutes) - container restarts can be normal, frequent restarts are not.

Reply:
d) we may consider expanding the time function to support additional parameters, but I think the pattern for this particular codebundle will stick to minutes (unless you see any use case where something larger might work)

Reply:
-^ I've updated the function and moved it, hopefully it's more useful now for other functions/use-cases.